### PR TITLE
Implement toJson, deepCopy, stringEscape, etc.

### PR DESCRIPTION
**Pull request by 7hong13**
#### What this PR does / why we need it?
1. Implemented `toJson`, `deepCopy`, `toSortedJson` functions which were left as `TODO`.
(`toJS` is not implemented yet. It is only for testing, so I guess it is not urgent.)
2. Fixed some minor issues to match with `yorkie-js-sdk`.
3. Added some test codes.
#### Any background context you want to provide?
For `stringEscape` function and its test code, I referred to [iOS version of it](https://github.com/yorkie-team/yorkie-ios-sdk/pull/22)

#### What are the relevant tickets?

Fixes #

### Checklist

- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: yorkie/src/main/kotlin/dev/yorkie/document/crdt/CrdtElement.kt [hunk 1]
```diff
@@ -32,7 +32,7 @@ internal abstract class CrdtElement(
 
     var removedAt: TimeTicket? = removedAt
         set(value) {
-            if (value > createdAt && value > field) {
+            if (value >= createdAt && value > field) {
```
Comment (7hong13) [inline yorkie/src/main/kotlin/dev/yorkie/document/crdt/CrdtElement.kt:35]: @skhugh this condition has some issues when setting dummyHead in RgaTreeList, so I fixed it.

File: yorkie/src/main/kotlin/dev/yorkie/util/SplayTreeSet.kt [hunk 2]
```diff
@@ -50,7 +50,7 @@ internal class SplayTreeSet<V>(lengthCalculator: LengthCalculator<V>? = null) {
         }
         var node: Node<V> = root
         while (true) {
-            if (target <= node.leftWeight) {
+            if (node.left != null && target <= node.leftWeight) {
```
Comment (hackerwins) [inline yorkie/src/main/kotlin/dev/yorkie/util/SplayTreeSet.kt:53]: 👍

File: yorkie/src/main/kotlin/dev/yorkie/document/crdt/CrdtElement.kt (hunk 1 above)
Comment (skhugh) [reply yorkie/src/main/kotlin/dev/yorkie/document/crdt/CrdtElement.kt:35]: I actually made custom setter to replace `remove` function.
we should delete one of them.

File: yorkie/src/main/kotlin/dev/yorkie/document/crdt/CrdtRoot.kt [hunk 3]
```diff
@@ -46,7 +47,7 @@ internal class CrdtRoot(val rootObject: CrdtObject) {
         val subPaths = mutableListOf<String>()
         while (pair.parent != null) {
```
Comment (skhugh) [inline yorkie/src/main/kotlin/dev/yorkie/document/crdt/CrdtRoot.kt:48]: how about chaning like below to remove ugly `!!`s :)
```suggestion
        while (true) {
            val parent = pair.parent ?: break
```

File: yorkie/src/main/kotlin/dev/yorkie/document/json/EscapeString.kt [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package dev.yorkie.document.json
+
+/**
+ * Escapes string.
+ */
+fun escapeString(str: String): String {
+    return str.map {
```
Comment (skhugh) [inline yorkie/src/main/kotlin/dev/yorkie/document/json/EscapeString.kt:7]: I think `kotlin.text.replace` is the better option here.

**Comment by skhugh** [comment]
I think `toJS` is not needed for Android SDK.
also `toSortedJson` is used for test code only so I prefer removing them (there should be simpler ways to check equality of JSON contents in Kotlin)

**Comment by 7hong13** [comment]
@hackerwins as I mentioned in the daily scrum, I removed `toSortedJson`. I will revert it if removing them can cause some issues.

**Review by hackerwins — APPROVED** [review]
Thanks for your contribution.

**Pull request merged**